cartography/intel/gcp/compute.py

```diff
@@ -553,6 +625,113 @@ def load_gcp_subnets(neo4j_session, subnets, gcp_update_tag):
         )
 
 
+@timeit
+def load_gcp_forwarding_rules(neo4j_session, fwd_rules, gcp_update_tag):
+    """
+    Ingest GCP forwarding rules data to Neo4j
+    :param neo4j_session: The Neo4j session
+    :param fwd_rules: List of forwarding rules
+    :param gcp_update_tag: The timestamp to set these Neo4j nodes with
+    :return: Nothing
+    """
+
+    query = """
+        MERGE(fwd:GCPForwardingRule{id:{PartialUri}})
+        ON CREATE SET fwd.firstseen = timestamp(),
+        fwd.partial_uri = {PartialUri}
+        SET fwd.ip_address = {IPAddress},
+        fwd.ip_protocol = {IPProtocol},
+        fwd.load_balancing_scheme = {LoadBalancingScheme},
+        fwd.name = {Name},
+        fwd.network = {NetworkPartialUri},
+        fwd.port_range = {PortRange},
+        fwd.ports = {Ports},
+        fwd.project_id = {ProjectId},
+        fwd.region = {Region},
+        fwd.self_link = {SelfLink},
+        fwd.subnetwork = {SubNetworkPartialUri},
+        fwd.target = {TargetPartialUri},
+        fwd.lastupdated = {gcp_update_tag}
+    """
+
+    for fwd in fwd_rules:
+        network = fwd.get('network', None)
+        subnetwork = fwd.get('subnetwork', None)
+
+        neo4j_session.run(
+            query,
+            PartialUri=fwd['partial_uri'],
+            IPAddress=fwd['ip_address'],
+            IPProtocol=fwd['ip_protocol'],
+            LoadBalancingScheme=fwd['load_balancing_scheme'],
+            Name=fwd['name'],
+            Network=network,
+            NetworkPartialUri=fwd.get('network_partial_uri', None),
+            PortRange=fwd.get('port_range', None),
+            Ports=fwd.get('ports', None),
+            ProjectId=fwd['project_id'],
+            Region=fwd.get('region', None),
+            SelfLink=fwd['self_link'],
+            SubNetwork=subnetwork,
+            SubNetworkPartialUri=fwd.get('subnetwork_partial_uri', None),
+            TargetPartialUri=fwd['target'],
+            gcp_update_tag=gcp_update_tag,
+        )
+
+        if subnetwork:
+            _attach_fwd_rule_to_subnet(neo4j_session, fwd, gcp_update_tag)
+        elif network:
+            _attach_fwd_rule_to_vpc(neo4j_session, fwd, gcp_update_tag)
+
+
+@timeit
+def _attach_fwd_rule_to_subnet(neo4j_session, fwd, gcp_update_tag):
+    query = """
+        MERGE(subnet:GCPSubnet{id:{SubNetworkPartialUri}})
+        ON CREATE SET subnet.firstseen = timestamp(),
+        subnet.partial_uri = {SubNetworkPartialUri}
+        SET subnet.lastupdated = {gcp_update_tag}
+
+        WITH subnet
+        MATCH(fwd:GCPForwardingRule{id:{PartialUri}})
+
+        MERGE(subnet)-[p:RESOURCE]->(fwd)
+        ON CREATE SET p.firstseen = timestamp()
+        SET p.lastupdated = {gcp_update_tag}
+    """
+
+    neo4j_session.run(
+        query,
+        PartialUri=fwd['partial_uri'],
+        SubNetworkPartialUri=fwd.get('subnetwork_partial_uri', None),
+        gcp_update_tag=gcp_update_tag,
+    )\
```

Comment:
Nit: does the training '\' cause problems?

Reply:
+1 on this